### DISCUSSION TITLE
Minified version at a smaller size

### DISCUSSION THREAD

**Discussion by GeorgianStan** [fabricjs/fabric.js#7061] [Q&A]
Hi there!

The current minified version of **fabric.js** has **300 KB**. Is there something smaller that we can use?

I mention that I'm using the library like this `import { fabric } from "fabric";` through Webpack.

**Comment by asturur** [comment]
The only way to make it smaller is cut out the parts you do not use.

Do you use svg export/import? if not you can remove it.
Do you use brushes? if not you can remove it.
Filters? removable.

Look at the build script, what i did was defining a post install script for my project.
In this post install script i go in node_modules/fabric and i run the build script removing the modules i do not need and you should get a smaller fabricJS.

**Reply by GeorgianStan** [reply]
Will do that. Thank you!

**Reply by asturur** [reply]
In order to get an hoc include we need to rewrite all the code to take advantage of es2018+ or wherever are we now, and let webpack make the magic.

**Reply by GeorgianStan** [reply]
Does it require `uglifyjs` to be installed as global package?

I received the following output:

```bash
λ node build.js
Built distribution to dist/fabric.js
Minification failed using uglifyjs with uglifyjs  --compress --mangle --output fabric.min.js fabric.js  
Minifier error output:
Error: Command failed: uglifyjs  --compress --mangle --output fabric.min.js fabric.js
'uglifyjs' is not recognized as an internal or external command,
operable program or batch file.
```

**Reply by asturur** [reply]
i think it should work. maybe is a dev-dep and is not installed by default in your case?
